app/main/views/reset_password.py

```diff
@@ -127,8 +127,29 @@ def send_reset_password_email():
                     }
                 )
         else:
+            # Send a email to the Notify sandbox using the 'inactive' template, to mitigate any timing attacks (where
+            # a user's existence could be determined by the response time of this view). Any errors are also handled
+            # in the same way as for inactive users.
+            try:
+                notify_client.send_email(
+                    NOTIFY_SANDBOX_ADDRESS,
+                    template_name_or_id=current_app.config['NOTIFY_TEMPLATES']['reset_password_inactive'],
+                    reference='reset-password-nonexistent-user-{}'.format(hash_string(email_address)),
+                )
+            except EmailError as exc:
+                log_email_error(
+                    exc,
+                    "Password reset (non-existent user)",
+                    "login.reset-email-nonexistent.notify-error",
+                    email_address,  # Hashed by the helper function
+                )
+                return render_template(
+                    'toolkit/errors/500.html',
+                    error_message="Failed to send password reset.",
+                ), 503
+
             current_app.logger.info(
-                "{code}: Password reset request for invalid email_hash {email_hash}",
+                "{code}: Sending password (non-)reset email for invalid user email_hash {email_hash}",
```

Comment:
_Technically_ it's already been sent.

Reply:
Good point!

Reply:
On closer inspection, we say the same thing for the other scenarios (valid user, invalid user). The request has been fired off to Notify, but the email itself probably hasn't been sent yet 😸 Shall we keep it consistent?

Reply:
:man_shrugging: I'd just change them all - there's never any point that we _know_ our email's been sent really - it could always be spooling in some midway mail server somewhere.

Reply:
Fair enough - fixed.